src/dstore/pmix_esh.c

```diff
@@ -0,0 +1,1374 @@
+    _update_initial_segment_info();
+
+    /* get information about shared segments per this namespace from the initial segment. */
+    ns_info = _get_ns_info_from_initial_segment(nspace);
+    if (NULL == ns_info) {
+        /* no data for this namespace is found in the shared memory. */
+        PMIX_OUTPUT_VERBOSE((7, pmix_globals.debug_output,
+                    "%s:%d:%s:  no data for ns %s is found in the shared memory.", __FILE__, __LINE__, __func__, nspace));
+        /* unset lock */
+        flock(lockfd, LOCK_UN);
+        return PMIX_ERROR;
+    }
+
+    /* get ns_track_elem_t object for the target namespace from the local track list. */
+    elem = _get_track_elem_for_namespace(nspace);
+    if (NULL == elem) {
+        PMIX_ERROR_LOG(PMIX_ERROR);
+        /* unset lock */
+        flock(lockfd, LOCK_UN);
+        return PMIX_ERROR;
+    }
+    /* need to update tracker:
+     * attach to shared memory regions for this namespace and store its info locally
+     * to operate with address and detach/unlink afterwards. */
+    rc = _update_ns_elem(elem, ns_info);
+    if (PMIX_SUCCESS != rc) {
+        PMIX_ERROR_LOG(PMIX_ERROR);
+        /* unset lock */
+        flock(lockfd, LOCK_UN);
+        return PMIX_ERROR;
+    }
+
+    /* now we know info about meta segment for this namespace. */
+    meta_seg = elem->meta_seg;
+    data_seg = elem->data_seg;
+
+    while (nprocs--) {
+        if (PMIX_RANK_UNDEF == rank) {
+            cur_rank++;
+        }
+        /* Then we look for the rank meta info in the shared meta segment. */
+        rinfo = _get_rank_meta_info(cur_rank, meta_seg);
+        if (NULL == rinfo) {
+            PMIX_OUTPUT_VERBOSE((7, pmix_globals.debug_output,
+                        "%s:%d:%s:  no data for this rank is found in the shared memory. rank %d", __FILE__, __LINE__, __func__, cur_rank));
+            rc = PMIX_ERR_PROC_ENTRY_NOT_FOUND;
+            continue;
+        }
+        addr = _get_data_region_by_offset(data_seg, rinfo->offset);
+        if (NULL == addr) {
+            PMIX_ERROR_LOG(PMIX_ERROR);
+            rc = PMIX_ERR_PROC_ENTRY_NOT_FOUND;
+            continue;
+        }
+        kval_cnt = rinfo->count;
+        /* TODO: probably PMIX_ERR_NOT_FOUND is a better way but
+         * setting to one initiates wrong next logic for unknown reason */
+        rc = PMIX_ERROR;
+
+        while (0 < kval_cnt) {
+            /* data is stored in the following format:
+             * key[PMIX_MAX_KEYLEN+1]
+             * size_t size
+             * byte buffer containing pmix_value, should be loaded to pmix_buffer_t and unpacked.
+             * next kval pair
+             * .....
+             * EXTENSION slot which has key = EXTENSION_SLOT and a size_t value for offset to next data address for this process.
+             */
+            if (0 == strncmp((const char *)addr, "INVALIDATED", PMIX_MAX_KEYLEN+1)) {
```

Comment:
@igor-ivanov 
* Define a constant for "INVALIDATED"
* Use PMIX_INT_INVALIDATED for safety.